Sources/TokamakCore/App/Scenes/SceneStorage.swift

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/17/20.
+//
+
+import OpenCombine
+
+public protocol _SceneStorageProvider {
```

Comment:
**suggestion:** since `_SceneStorageProvider` and `_AppStorageProvider` are nearly identical, could this be a single protocol shared in both cases?